RecoTracker/MkFitCMS/src/MkStdSeqs.cc

```diff
@@ -0,0 +1,586 @@
+                  if (hitidx2 >= 0) {
+                    auto it = std::find_if(track.BeginHitsOnTrack(),
+                                           track.EndHitsOnTrack(),
+                                           [&hitidx2, &hitlyr2](const HitOnTrack &element) {
+                                             return (element.index == hitidx2 && element.layer == hitlyr2);
+                                           });
+                    if (it != track.EndHitsOnTrack())
+                      numHitsShared++;
+                  }
+                }
+
+                float fracHitsShared = numHitsShared / std::min(track.nFoundHits(), track2.nFoundHits());
+                //Only remove one of the tracks if they share at least X% of the hits (denominator is the shorter track)
+                if (fracHitsShared < Config::minFracHitsShared)
+                  continue;
+              }
+              //Keep track with best score
+              if (track.score() > track2.score())
+                track2.setDuplicateValue(true);
+              else
+                track.setDuplicateValue(true);
+            }  //end of if dPt
+          }    //end of else
+        }      //end of loop over track2
+      }        //end of loop over track1
+    }
+
+    void remove_duplicates(TrackVec &tracks) {
+      tracks.erase(std::remove_if(tracks.begin(), tracks.end(), [](auto track) { return track.getDuplicateValue(); }),
+                   tracks.end());
+    }
+
+    //=========================================================================
+    // SHARED HITS DUPLICATE CLEANING
+    //=========================================================================
+
+    void find_duplicates_sharedhits(TrackVec &tracks, const float fraction) {
+      const auto ntracks = tracks.size();
+
+      std::vector<float> ctheta(ntracks);
+      for (auto itrack = 0U; itrack < ntracks; itrack++) {
+        auto &trk = tracks[itrack];
+        ctheta[itrack] = 1.f / std::tan(trk.theta());
+      }
+
+      for (auto itrack = 0U; itrack < ntracks; itrack++) {
+        auto &trk = tracks[itrack];
+        auto phi1 = trk.momPhi();
+        auto ctheta1 = ctheta[itrack];
+
+        for (auto jtrack = itrack + 1; jtrack < ntracks; jtrack++) {
+          auto &track2 = tracks[jtrack];
+          auto sharedCount = 0;
+          auto sharedFirst = 0;
+
+          auto dctheta = std::abs(ctheta[jtrack] - ctheta1);
+          if (dctheta > 1.)
+            continue;
+
+          auto dphi = std::abs(squashPhiMinimal(phi1 - track2.momPhi()));
+          if (dphi > 1.)
+            continue;
+
+          for (int i = 0; i < trk.nTotalHits(); ++i) {
+            if (trk.getHitIdx(i) < 0)
+              continue;
+            int a = trk.getHitLyr(i);
+            int b = trk.getHitIdx(i);
+            for (int j = 0; j < track2.nTotalHits(); ++j) {
+              if (track2.getHitIdx(j) < 0)
+                continue;
+              int c = track2.getHitLyr(j);
```

Comment:
const here and below

Reply:
Addressed in Round 1 follow-up PR.
trackreco/cmssw#75 https://github.com/trackreco/cmssw/pull/75/commits/e9834b4a33dab0a34078a4861b825f0e1ebdb64d